### ISSUE TITLE
Setting default font-size

### ISSUE THREAD

**Issue by os1r1s110** [wkhtmltopdf/wkhtmltopdf#3569]
I'd like to know if it's possible to change the default font-size. It would be really useful as I'm trying to use bootstrap 4 which uses rem units by default for all the typography (titles, headers, paragraphs, etc...) but that heavily relies on the base font-size which is usually set to 16px.

I'd like to be able to specify another default font-size as being the default one (in hope that rem units would pick it up).

I found by searching that Qt had an option for that in QWebSettings.DefaultFontSize but it doesn't seem available through the WkHtmlToPdf library.... is there a specific reason why?

Thanks in advance!

**Comment by naggie** [comment]
Hm, I just experienced exactly the same thing within hours. Thanks for reporting.

**Comment by os1r1s110** [comment]
@naggie As of now, I made a "hacky workaround". If you are building from source, you can add a custom scss variable, for example $remEquivalent and set it to x px. Then find and replace all occurences of the "rem" unit in your custom variables file and replace it with "* $remEquivalent".

This way, you will be able to only change this single variable and recompile your css from it. The real downside though is that you cannot change it dynamically when calling the tool... so that might not be a good solution on the long term, I wish someone will have a more robust solution :P

**Comment by naggie** [comment]
Thanks -- yes I'm using bootstrap 4. I'll give it a go!

**Comment by naggie** [comment]
I found running `wkhtmltopdf --dpi 300 --zoom 2` worked around the issue, possibly causes by https://github.com/wkhtmltopdf/wkhtmltopdf/issues/3241 

...however for me the bootstrap grids are now strangely overlapping on one line.....
![image](https://user-images.githubusercontent.com/208440/28394221-4290d5a0-6ce3-11e7-9d28-a441b050076c.png)

I bet it's unrelated.

**Comment by os1r1s110** [comment]
Wow that's weird.... these are your "col" ? If they are, I would suspect the webkit rendering engine to not support flexbox (which is now used by default by bootstrap).

**Comment by naggie** [comment]
Yes I guessed that may be the case and had already changed to the manual col-x. However now only one item renders at what looks like a very small screen width. I'll keep investigating. Thanks for the advice.

**Comment by os1r1s110** [comment]
Keep us in touch, I will surely fall into the same pitfalls (implementing styles tomorrow, I only did the setup of configuring WkHtml with a C# wrapper and pointing to the custom bootstrap 4 build for printing, but didn't create the "views to render as pdf" yet... :)

**Comment by os1r1s110** [comment]
Which version of WkHtmlToPdf are you using? I just tried with the latest "stable" version (0.12.4) and it seems to work with basic flexbox usage (rows, cols, seem to be respected)

To be noted, I use gulp to compile the scss to css and I have included Autoprefixer for which I specified "chrome >= 13", so that might be the reason why it seems to work.

Don't take it as granted though as I barely scratched the surface and tested very quickly but it seems promising :)

**Comment by naggie** [comment]
I was also using 0.12.4. I have got to the stage where everything renders fine, but the problem is that it prints only one column on the page, as if the effective screen width is rather narrow. I'm sure this is due to the `rem` units -- bootstrap 3 works fine.

I've yet to try your scss hack though -- it's a backup option.

**Comment by apphancer** [comment]
I've fixed the grid issue in my project by adding some custom CSS rules. Bootstrap 4 was not floating the grid cells.
I took some code from Bootstrap 3 and added it in a custom css file. What the custom css does is to add float: left to the col classes (these are the classes that are used when rendering with wkhtmltopdf) and also added some float clearing.
```
/* start - fix Bootstrap 4 issue in WKHTMLTOPDF */
.col-1, .col-2, .col-3, .col-4,  .col-5, .col-6, .col-7, .col-8, .col-9, .col-10, .col-11, .col-12 {
  float: left;
}
.row:before,
.row:after {
  display: table;
  content: " ";
}

.row:after {
  clear: both;
}

.row:before,
.row:after {
  display: table;
  content: " ";
}

.row:after {
  clear: both;
}
/* end - fix Bootstrap 4 issue in WKHTMLTOPDF */
```
